### PR TITLE
Workaround editor jumping on inserting non-snippet text

### PR DESCRIPTION
**Pull request by zardoy**
You must probably noticed that editors jumps higher when you apply postfix on multiline expression.
I'm pretty sure this behavior is annoying for all users, so there is workaround for non-snippets at least.

Attaching some gifs for context.

By using empty insert text we can workaround this behavior:

### CODE REVIEW

File: src/completionItemBuilder.ts [hunk 1]
```diff
@@ -35,13 +35,9 @@ export class CompletionItemBuilder {
...
     this.addCodeBlockDescription(this.replaceExpression(replacement, this.code))
-
-    if (useSnippets) {
-      const escapedCode = this.code.replace(/\$/g, '\\$')
-
-      this.item.insertText = new vsc.SnippetString(this.replaceExpression(replacement, escapedCode));
-    } else {
-      this.item.insertText = this.replaceExpression(replacement, this.code)
+    // don't have unscaped $ in replacement? not a snippet.
+    if (useSnippets && !/(?<!\\)\$/.test(replacement)) {
```
Comment (zardoy) [inline src/completionItemBuilder.ts:39]: I added this to support simple custom postfixes, like `await` from gif

**Comment by zardoy** [comment]
Here is the same fix for snippets, it just works in insiders and can be merged when new version is released: https://github.com/zardoy/vscode-postfix-ts/compare/fix-editor-jumping...zardoy:vscode-postfix-ts:fix-editor-jumping-snippets

**Comment by ipatalas** [comment]
Hi, thanks for that. I didn't even notice it's jumping all over the place but I haven't been using TS/JS (and this extension) for quite some time so that might be why ;)

However I decided to go with little bit different approach for custom templates.
That hardcoded `useSnippet` parameter was good at the beginning because it was easy to introduce that but now I think it's just better to auto-detect that so I removed that parameter: https://github.com/ipatalas/vscode-postfix-ts/commit/8b6d9c8b0d0ad11deff09b40ccc6fdc1ea0d7832
The rest of the change is pretty much yours so you get the credit for that :)
Also this should work well when this `vsc.SnippetTextEdit` will be introduced, just set `insertText` to empty and add `vsc.SnippetTextEdit.replace` to additional edits.

**Comment by zardoy** [comment]
Hey, @ipatalas thanks for linking the commit!

I've reviewed the changes and can say for sure that `/(?<!\\)\$\{?\d/` regex breaks usage with [variables](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_variables) defined in templates. For example I could use this body: `console.log('$RELATIVE_FILEPATH:$TM_LINE_NUMBER', {{expr}})` -> `console.log('test/check.ts:1', expr)`

As I understand from [VSCode docs](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_how-do-i-have-a-snippet-place-a-variable-in-the-pasted-script), my regex seems to be correct.

**Comment by ipatalas** [comment]
> I've reviewed the changes and can say for sure that /(?<!\\)\$\{?\d/ regex breaks usage with [variables](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_variables) defined in templates. For example I could use this body: console.log('$RELATIVE_FILEPATH:$TM_LINE_NUMBER', {{expr}}) -> console.log('test/check.ts:1', expr)

Not exactly true ;)

Bear in mind `\$\{?\d` needs to end with a digit so only things like `$1` or `${1}` will match.
Check the results (no match for this example):
image

**Comment by zardoy** [comment]
> Not exactly true ;)

Hey! Im still confused. Even screenshot shows to you that `body` I provided above now is not recongnized as snippet, but `$RELATIVE_FILEPATH` should be matched since its a snippet variable e.g. we must use insert snippet in this case.

I checked that after last commit my postfix is being inserted is regular text like `console.log('$RELATIVE_FILEPATH:$TM_LINE_NUMBER', expr)` but it should be inserted as `console.log('test/check.ts:1', expr)`.

If you still don't understand what I mean just try to create postfix with the following `body`:

```txt
console.log('$RELATIVE_FILEPATH:$TM_LINE_NUMBER', {{expr}})
```

and compare insertText with your and mine regex. Feel free to ask if something isn't clear :)

**Comment by ipatalas** [comment]
Oh, sorry. You're right. I got it wrong, my bad.
Luckily that's easy to fix, thanks for making that clear.

**Comment by zardoy** [comment]
https://github.com/ipatalas/vscode-postfix-ts/pull/75

**Pull request closed without merging**